plugins/inputs/sqlserver/azuresqlqueries.go

```diff
@@ -679,6 +679,138 @@ SELECT
+	MAX(rs.max_query_max_used_memory) AS max_query_max_used_memory,
+	SUM(rs.avg_query_max_used_memory * rs.count_executions) / SUM(rs.count_executions) AS avg_query_max_used_memory,
+	MAX(rs.max_duration) as [max_duration],
+	SUM(rs.avg_duration * rs.count_executions) / SUM(rs.count_executions) as avg_duration,
+	MAX(rs.max_num_physical_io_reads) as max_num_physical_io_reads,
+	SUM(rs.avg_num_physical_io_reads * rs.count_executions) / SUM(rs.count_executions) as avg_num_physical_io_reads,
+	MAX(rs.max_page_server_io_reads) as max_page_server_io_reads,
+	SUM(rs.avg_page_server_io_reads * rs.count_executions) / SUM(rs.count_executions) as avg_page_server_io_reads,
+	MAX(rs.max_log_bytes_used) as [max_log_bytes_used],
+	SUM(rs.avg_log_bytes_used * rs.count_executions) / SUM(rs.count_executions) as avg_log_bytes_used
+FROM sys.query_store_query q WITH (NOLOCK)
+	JOIN sys.query_store_plan p WITH (NOLOCK) ON q.query_id = p.query_id
+	JOIN sys.query_store_runtime_stats rs WITH (NOLOCK) ON p.plan_id = rs.plan_id
+	JOIN sys.query_store_runtime_stats_interval rsi WITH (NOLOCK) ON rs.runtime_stats_interval_id = rsi.runtime_stats_interval_id
+WHERE rsi.start_time >= @currIntervalStartDate AND rsi.end_time <= @currIntervalEndDate
+GROUP BY rs.plan_id, rs.execution_type, rs.runtime_stats_interval_id;
+
+/* Return the timestamp when the query was run, to be used in the next call */
+SELECT CAST(@currIntervalEndTimestamp AS nvarchar(35)) AS QueryData;
+`
+
+const sqlAzureDBQueryStoreWaitStatistics = sqlAzureDBPartQueryPeriod + `
+
+/*QDS is not enabled on read-only replicas: https://docs.microsoft.com/en-us/sql/t-sql/functions/databasepropertyex-transact-sql?view=sql-server-ver15*/
+IF DATABASEPROPERTYEX(DB_Name(), 'Updatability') = 'READ_ONLY'  
+	RETURN
+
+/* Query the data from query store. Group  by plan_id, runtime_stats_interval_id, execution_type and wait_category, and specified here:
+https://docs.microsoft.com/en-us/sql/relational-databases/system-catalog-views/sys-query-store-wait-stats-transact-sql?view=sql-server-ver15
+*/
+SELECT
+	'sqlserver_azuredb_querystore_wait_stats' AS [measurement],
+	REPLACE(@@SERVERNAME, '\', ':') AS [sql_instance],
+	DB_NAME() AS [database_name],
+	CONVERT(nvarchar(35), MAX(rsi.start_time), 126) AS interval_start_time,
+	CONVERT(nvarchar(35), MAX(rsi.end_time), 126) AS interval_end_time,
+	CAST(MAX(q.query_id) AS nvarchar(20)) AS query_id,
+	CONVERT(nvarchar(20), MAX(q.query_hash), 1) as query_hash,
+	CAST(ws.plan_id AS nvarchar(20)) AS plan_id,
+	CONVERT(nvarchar(20), MAX(p.query_plan_hash), 1) as query_plan_hash,
+	ws.wait_category_desc AS wait_category_s,
+	ws.execution_type AS exec_type_d,
+	MAX(ws.execution_type_desc) AS exec_type_desc,
+	SUM(ws.total_query_wait_time_ms) AS total_query_wait_time_ms_d,
+	CAST(SUM(IIF(avg_query_wait_time_ms = 0, 0, ws.total_query_wait_time_ms / avg_query_wait_time_ms)) AS BIGINT) AS count_executions,
```

Comment:
This is a very imprecise way to calculate the number of executions, each execution can wait for drastically different durations. Why do we need count of executions here?

Reply:
@dimitri-furman if this calculation is imprecise, what can you suggest instead? This field is necessary for Azure SQL Insights solution.

Reply:
To get the precise count, you would need to aggregate runtime stats in the same way as you are aggregating waits, and then join the two result sets. But before we do this, I'd like to understand how this column is used in SQL Insights. Let's discuss offline.

Reply:
@dimitri-furman, in order to not confuse with **count_executions** from QDS runtime stats, the column was renamed to **count_waits**.

Reply:
This is still problematic I am afraid. For one thing, "count" implies that the value is an integer, but this is fractional. For another, "wait" has a very specific meaning in the context of MSSQL query execution, i.e. see the definition of `waiting_tasks_count` in [sys.dm_os_wait_stats](https://docs.microsoft.com/en-us/sql/relational-databases/system-dynamic-management-views/sys-dm-os-wait-stats-transact-sql). This new name makes it look like this is the same thing, but it's actually something quite different and artificial.

If this is only needed for SQL Insights, then I don't think we should have this column in this query at all. The data we need is already collected in the runtime stats query for each query hash (`count_executions`), so a calculation involving waits for a query hash should be done in KQL, not as part of collection.

Reply:
@dimitri-furman, I'am afraid that using of count_executions (taken from sys.query_store_runtime_stats) leads to wrong results in most cases, because the same query may face with different wait types when executed multiple times, i.e. a query may be executed 10 times but it may face with memory waits only 6 times and this value then used to calculate wait statistics in QDS. This field cannot be calculated in KQL after all aggregations in the telegraf's query. Without this field we won't be able to calculate correct average values for wait statistics in KQL. And this count is not fractional, when you divide total on avg (both taken from **sys.query_store_wait_stats**) you will almost always receive integer (or smth. very close to integer in rare cases) because this integer value is used to calculate the avg. And I don't know why **sys.query_store_wait_stats** doen't contain any field containing count, it looks like an issue. Azure SQL DB QDS wait stats metrics (can be enabled by turning on **QueryStoreWaitStatistics** in Diagnostic settings) contain count. And often it differs from **count_executions**.
I don't understand, why you mentioned **sys.dm_os_wait_stats**. The field you mentioned has different name, it is located in another view and therefore has different meaning. These fields can have similar but not the same meaning.

Reply:
I checked with a QDS SME and confirmed that we account for every query execution when calculating `avg_query_wait_time_ms`, not just the ones where the query actually waited, which was my assumption. So the current calculation in the query does work, and produces an accurate number of query executions per interval *and wait category* (so it won't usually match execution count per interval from runtime stats). 

This means that the original column name, `count_executions`, was actually accurate. `count_waits` is a bit misleading because it implies that we are counting the number of waits, which can be more or less than the number of executions (a query could wait multiple times during execution, and some executions may not wait at all as you pointed out). I think we should revert this name change.

Apologies for the confusion and runaround.

Minor style nit: it should be `ws.avg_query_wait_time_ms`, not `avg_query_wait_time_ms`, in two places.

Reply:
Fixed